pyogrio/_io.pyx

```diff
@@ -1679,13 +1679,11 @@ cdef infer_field_types(list dtypes):
     return field_types
 
 
-# TODO: set geometry and field data as memory views?
-def ogr_write(
-    str path, str layer, str driver, geometry, fields, field_data, field_mask,
-    str crs, str geometry_type, str encoding, object dataset_kwargs,
-    object layer_kwargs, bint promote_to_multi=False, bint nan_as_null=True,
-    bint append=False, dataset_metadata=None, layer_metadata=None,
-    gdal_tz_offsets=None
+cdef create_dataset_layer(
```

Comment:
Can you please add a minimal docstring to this to indicate it returns a boolean to indicate that a new layer is created in the process (vs append to existing), that it sets `ogr_dataset` and `ogr_layer` pointers with valid objects on success and raise exceptions otherwise, and that it is then the responsibility of the caller to release those?

Reply:
Sure!